src/libraries/System.IO.FileSystem/tests/Directory/CreateTempSubdirectory.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Directory_CreateTempSubdirectory : FileSystemTest
+    {
+        public static TheoryData<string> CreateTempSubdirectoryData
+        {
+            get
+            {
+                var result = new TheoryData<string>() { null, "", "myDir", "my.Dir" };
+                if (!OperatingSystem.IsWindows())
+                {
+                    // ensure we can use backslashes on Unix since that isn't a directory separator
+                    result.Add(@"my\File");
+                    result.Add(@"\");
+                }
+                return result;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(CreateTempSubdirectoryData))]
+        public void CreateTempSubdirectory(string prefix)
+        {
+            DirectoryInfo tmpDir = Directory.CreateTempSubdirectory(prefix);
+            try
+            {
+                Assert.True(tmpDir.Exists);
+                Assert.Equal(-1, tmpDir.FullName.IndexOfAny(Path.GetInvalidPathChars()));
+                Assert.Empty(Directory.GetFiles(tmpDir.FullName));
+                Assert.Equal(Path.TrimEndingDirectorySeparator(Path.GetTempPath()), tmpDir.Parent.FullName);
+
+                if (!string.IsNullOrEmpty(prefix))
+                {
+                    Assert.StartsWith(prefix, tmpDir.Name);
+                }
```

Comment:
Not very important but could we check that the prefix does not affect the unique name length? i.e: it keeps being 6 on Unix and 12 on Windows.